controllers/organization_controller_sync.go

```diff
@@ -0,0 +1,123 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	orgv1 "github.com/appuio/control-api/apis/organization/v1"
+	controlv1 "github.com/appuio/control-api/apis/v1"
+	"github.com/vshn/appuio-keycloak-adapter/keycloak"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var orgImportAnnot = "keycloak-adapter.vshn.net/importing"
+
+// Sync lists all Keycloak groups in the realm and creates corresponding Organizations if they do not exist
+func (r *OrganizationReconciler) Sync(ctx context.Context) error {
+	log := log.FromContext(ctx)
+
+	gs, err := r.Keycloak.ListGroups(ctx)
+	if err != nil {
+		log.Error(err, "error listing Keycloak groups")
+		return err
+	}
+
+	orgMap, err := r.getOrganiztionMap(ctx)
+	if err != nil {
+		log.Error(err, "error listing Organizations")
+		return err
+	}
+
+	var groupErr error
+	recordError := func(descr string, g keycloak.Group, err error) {
+		if groupErr == nil {
+			groupErr = errors.New("")
+		}
+		log.WithValues("group", g).Error(err, descr)
+		groupErr = fmt.Errorf("%w\n%s: %s\n    %s", groupErr, g.Name, descr, err.Error())
+	}
```

Comment:
We might also look into https://github.com/hashicorp/go-multierror instead

Reply:
Hmm maybe that would be a bit more readable. But as I refactored error handling in https://github.com/vshn/appuio-keycloak-adapter/pull/8 anyway I would leave this as is for this PR

Reply:
ah, wasn't aware of that. Yes, events would be the better option